.github/workflows/cygwin-cmake.yml

```diff
@@ -0,0 +1,50 @@
+name: hdf5 dev CMake Cygwin
+
+on:
+  workflow_call:
+    inputs:
+      build_mode:
+        description: "release vs. debug build"
+        required: true
+        type: string
+
+jobs:
+  cygwin_build_and_test:
+    name: "Cygwin ${{ inputs.build_mode }}"
+    runs-on: windows-latest
+    steps:
+      - name: Set git to use LF
+        run: |
+          git config --global core.autocrlf input
+
+      - uses: actions/checkout@v4
+
+      - name: Install Cygwin
+        uses: cygwin/cygwin-install-action@master
+        with:
+          packages: cmake gcc-fortran make ninja perl zlib-devel
+
+      - name: Configure
+        run: |
+          export PATH=/usr/bin:$PATH
+          mkdir build
+          cd build
+          cmake -G Ninja \
+          -DCMAKE_BUILD_TYPE=${{ inputs.build_mode }} \
+          -D BUILD_SHARED_LIBS:BOOL=OFF \
+           ..
+        shell: C:\cygwin\bin\bash.exe -eo pipefail -o igncr '{0}'
+
+      - name: Build
+        run: |
+          export PATH=/usr/bin:$PATH
+          cd build
+          ctest -T Build
+        shell: C:\cygwin\bin\bash.exe -eo pipefail -o igncr '{0}'
+
+      - name: Test
+        run: |
+          export PATH=/usr/bin:$PATH
+          cd build
+          ctest -T Test
```

Comment:
```suggestion
         ctest --test-dir build -T Test
```

Reply:
@tbeu , thanks for taking the time to review my pull request! I also wanted to express my sincere gratitude for your work on maintaining the [matio](https://github.com/tbeu/matio) project. It's a valuable resource for the community. I'm happy to see that Cygwin is now a part of matio CI.

Reply:
Sure, its give and take. ❤️